vladiate/validators.py

```diff
@@ -121,10 +121,12 @@ def bad(self):
 class RegexValidator(Validator):
     """ Validates that a field matches a given regex """
 
-    def __init__(self, pattern=r"di^", **kwargs):
+    def __init__(self, pattern=r"di^", full=False, **kwargs):
         super(RegexValidator, self).__init__(**kwargs)
         self.regex = re.compile(pattern)
         self.failures = set([])
+        if full:
+            self.regex = re.compile(r"(?:" + pattern + r")\Z")
```

Comment:
Can you move the current compilation to an else statement so we aren't potentially compiling the regex twice?

```python
else:
    self.regex = re.compile(pattern)
```

Reply:
Sure, done.